pkg/provision/workspace/tokens.go

```diff
@@ -140,11 +140,12 @@ func checkSATokenPathCollisions(saTokens []v1alpha1.ServiceAccountToken, mountPa
 			for _, saToken := range collidingSATokens {
 				problemNames = append(problemNames, saToken.Name)
 			}
-
 			if len(problemPaths) == 1 {
+				sort.Strings(problemNames)
 				return fmt.Errorf("the following ServiceAccount tokens have the same path (%s) and mount path (%s): %s", path, mountPath, strings.Join(problemNames, ", "))
 			}
 		}
+		sort.Strings(problemNames)
 		return fmt.Errorf("multiple ServiceAccount tokens share the same path and mount path: %s", strings.Join(problemNames, ", "))
```

Comment:
Agreed. I've only encountered the CI tests failing once from the issue this is trying to fix.
We could just leave it out and see if it pops up in the future? :man_shrugging: 

Alternatively, the SA tokens test could be reworked a bit to prevent this flaky test (would involve using a regex to split up the error message into SA tokens, and then sort that result).

Reply:
Sounds good :)